test/driveTest.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:flutter_app/util/supabase.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'mockServer.dart';
+
+// DIe Klasse UrlProccessor muss zu begin jeder Testdatei implementiert werden und die Methode processUrl überschrieben werden
+// Wird die Methode ProcessUrl aufgrufen, wird für den dort definierten Fall (in dem Beispiel client.from('drives').select('driver_id,seats')) die Antwort definiert
+// Die Datenbankabfrage an den Client wird so abgefangen und das gefünschte ergebnis zurückgegeben.
+// Um herauszifinden welche URL durch die jeweilige Datenbankabfrage generiert wird, einfach den auskommentierten Print-Aufruf in der mockServer.Dart Datei aktivieren
+class DriveProccesor extends UrlProccessor {
```

Comment:
what if I want different test cases with different responses for the same request ? would I need to write a processor for each one or is there a simpler way?

Reply:
What you could do is Mock the method i guess that's probably for most cases easier

Reply:
I don't understand, what exactly would that look like?